gdpr/anonymizers/model_anonymizers.py

```diff
@@ -0,0 +1,349 @@
+
+    def get_deanonymized_value_from_obj(self, field: FieldAnonymizer, obj: Model, name: str) -> Any:
+        """Get from field, obj and field name deanonymized value."""
+        return field.get_deanonymized_value_from_obj(obj, name, self._get_encryption_key(obj, name))
+
+    def get_anonymized_value_from_version(self, field: FieldAnonymizer, obj: Model, version, name: str) -> Any:
+        """Get from field, obj and field name anonymized value."""
+        return field.get_anonymized_value_from_version(obj, version, name, self._get_encryption_key(obj, name))
+
+    def get_deanonymized_value_from_version(self, field: FieldAnonymizer, obj: Model, version, name: str) -> Any:
+        """Get from field, obj and field name deanonymized value."""
+        return field.get_deanonymized_value_from_version(obj, version, name, self._get_encryption_key(obj, name))
+
+    def _perform_anonymization(self, obj: Model, updated_data: dict,
+                               legal_reason: Optional[LegalReason] = None):
+        obj.__class__.objects.filter(pk=obj.pk).update(**updated_data)
+        for i in updated_data.keys():
+            self.mark_field_as_anonymized(obj, i, legal_reason)
+
+    def _perform_deanonymization(self, obj: Model, updated_data: dict):
+        obj.__class__.objects.filter(pk=obj.pk).update(**updated_data)
+        for i in updated_data.keys():
+            self.unmark_field_as_anonymized(obj, i)
+
+    def perform_anonymization(self, obj: Model, updated_data: dict,
+                              legal_reason: Optional[LegalReason] = None):
+        """Update data in database and mark them as anonymized."""
+        with transaction.atomic():
+            self._perform_anonymization(obj, updated_data, legal_reason)
+
+    def perform_deanonymization(self, obj: Model, updated_data: dict):
+        """Update data in database and mark them as anonymized."""
+        with transaction.atomic():
+            self._perform_deanonymization(obj, updated_data)
+
+    @staticmethod
+    def _perform_version_update(model: Type[Model], version, update_data):
+        version_dict_local = dict(version.field_dict)
+        version_dict_local.update(update_data)
+        local_obj = model(**version_dict_local)
+        version_options = _get_options(model)
+        version.serialized_data = serializers.serialize(
+            version_options.format,
+            (local_obj,),
+            fields=version_options.fields
+        )
+        version.save()
+
+    def perform_anonymization_with_version(self, obj: Model, updated_data: Dict[str, Any],
+                                           updated_version_data: List[Tuple[Any, Dict]],
+                                           legal_reason: Optional[LegalReason] = None):
+        """Update data in database and versions and mark them as anonymized."""
+        with transaction.atomic():
+            # first we need to update versions
+            for version, version_dict in updated_version_data:
```

Comment:
počítáš s tím, že reversion je vždy nainstalované?

Reply:
pokud uživatel to zapne v Meta tak počítám že je nainstalované